### ISSUE TITLE
[a11y]: Inside the Switcher we are using SwitcherDivider getting error Ensure that lists are structured correctly

### ISSUE THREAD

**Issue by Meena9** [carbon-design-system/carbon#20619]
### Package

@carbon/react

### Browser

Chrome, Safari, Firefox, Edge

### Operating System

Windows

### Package version

1.89.0

### React version

18.3.1

### Automated testing tool and ruleset

axe-core/playwright and Axe rule ID List

### Description

`<ul>` and `<ol>` must only directly contain `<li>, <script> or <template>` elements
Element location: `.cds--switcher `
Element source `<ul class="cds--switcher">`

### WCAG 2.1 Violation

List element has direct children that are not allowed: hr

### Reproduction/example

 `<Switcher expanded={showRightPanel}> <SwitcherDivider /> <SwitcherItem/></ Switcher>`

### Steps to reproduce

 add Switcher component then inside that add SwitcherDivider and also add SwitcherItem 

### Suggested Severity

Severity 2 = User cannot complete task, and/or no workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by nehaaprasad** [comment]
Hello! @alisonjoseph  and @Meena9 
I've fixed the SwitcherDivider to wrap the hr element in an li so it follows proper list structure and resolve.
let me know
